libs/langchain/langchain/embeddings/base.py

```diff
@@ -1,4 +1,226 @@
+        from langchain_openai import AzureOpenAIEmbeddings
+
+        return AzureOpenAIEmbeddings(model=model_name, **kwargs)
+    elif provider == "google_vertexai":
+        from langchain_google_vertexai import VertexAIEmbeddings
+
+        return VertexAIEmbeddings(model=model_name, **kwargs)
+    elif provider == "bedrock":
+        from langchain_aws import BedrockEmbeddings
+
+        return BedrockEmbeddings(model_id=model_name, **kwargs)
+    elif provider == "cohere":
+        from langchain_cohere import CohereEmbeddings
+
+        return CohereEmbeddings(model=model_name, **kwargs)
+    elif provider == "mistralai":
+        from langchain_mistralai import MistralAIEmbeddings
+
+        return MistralAIEmbeddings(model=model_name, **kwargs)
+    elif provider == "huggingface":
+        from langchain_huggingface import HuggingFaceEmbeddings
+
+        return HuggingFaceEmbeddings(model_name=model_name, **kwargs)
+    else:
+        raise ValueError(
+            f"Provider '{provider}' is not supported.\n"
+            f"Supported providers and their required packages:\n"
+            f"{_get_provider_list()}"
+        )
+
+
+@functools.lru_cache
+def _check_pkg(pkg: str) -> None:
+    """Check if a package is installed."""
+    if not util.find_spec(pkg):
+        raise ImportError(
+            f"Could not import {pkg} python package. "
+            f"Please install it with `pip install {pkg}`"
+        )
+
+
+def embedding_model(
+    model: str,
+    *,
+    model_provider: Optional[str] = None,
+    **kwargs: Any,
+) -> Union[Embeddings, Runnable[Any, List[float]]]:
+    f"""Initialize an embeddings model from a model name and optional provider.
+
+    This function creates an embeddings model instance from either:
+    1. A model string in the format 'provider:model-name'
+    2. A model name and explicit provider
+
+    Args:
+        model: Name of the model to use. Can be either:
+            - A model string like "openai:text-embedding-3-small"
+            - Just the model name if model_provider is specified
+        model_provider: Optional explicit provider name. If not specified,
+            will attempt to parse from the model string. Supported providers
+            and their required packages:
+
+            {_get_provider_list()}
+
+        **kwargs: Additional model-specific parameters passed to the embedding model.
+            These vary by provider, see the provider-specific documentation for details.
+
+    Returns:
+        An Embeddings instance that can generate embeddings for text.
+
+    Raises:
+        ValueError: If the model provider is not supported or cannot be determined
+        ImportError: If the required provider package is not installed
+
+    Examples:
+        >>> # Using a model string
+        >>> model = init_embedding_model("openai:text-embedding-3-small")
+        >>> model.embed_query("Hello, world!")
+        
+        >>> # Using explicit provider
```

Comment:
would double check that these render correctly in api ref, not sure they will

Reply:
wlil fix